serlib/ser_pattern.ml

```diff
@@ -36,6 +36,22 @@ type case_info_pattern =
 
 let hash_fold_array = hash_fold_array_frozen
 
+(** Can't properly serialize this GADT
+    Currently we only expose the derives for constr_pattern so can't occur anyway.
```

Comment:
Note that if that gets exposed in the future, people will get weird errors on the Ast; up to you, but maybe worth piercing instead, getting the Coq Ast is the main use case of SerAPI these days so I'm wary this could affect lots of ppl.